ros_buildfarm/templates/release/rpm/sync_packages_to_main_job.xml.em

```diff
@@ -43,32 +43,24 @@
   </triggers>
   <concurrentBuild>false</concurrentBuild>
   <builders>
+@{
+target_map = {}
+for os_name, os_code_name, arch in sync_targets:
+    target_map.setdefault(os_name, {})
+    target_map[os_name].setdefault(os_code_name, set())
+    target_map[os_name][os_code_name].add(arch)
+cra_sync_targets = []
+for os_name, os_code_names in target_map.items():
+    for os_code_name, arches in os_code_names.items():
+        cra_sync_targets.append((os_name, os_code_name, rosdistro_name, arches))
+}@
```

Comment:
This snip is necessary because there is a problem with the way we use EmPy and local/global variable access form within generator expressions. I tried to fix it in #594, but that effort never concluded.

Reply:
I found a way to get around this problem.
The short story is that we can't use template-level parameters within the body of generator expressions because they are not in scope. For example, assuming the template takes variables `foo` and `bar`, this will not work:
`['echo %s, %s!' % (foo, b) for b in bar]`
However, we can specifically ensure `foo` is accessible within the body by specifically passing it into the expression like this:
`['echo %s, %s!' % (foo, b) for foo, bar in ((foo, bar),) for b in bar]`